## Gradient Field Based Planner and Controller ##
GradFieldPlannerROS:
  attractor:
    search_dir_8: false # whether to use 8 search directions during update. Default: false

  general:
    R: 14
    end_pos_tol: 0.1 # Goal position tolerance in meters. Default: 0.1 [m]
    end_ang_tol: 0.1 # Goal orientation tolerance in rad. Default: 0.2 [rad]
    max_trans_vel: 0.22 # Maximal translational velocity in m/s. Default: 4 [m/s]
    max_trans_acc: 1.0 # Maximal translational acceleration in m/s². Default: 2 [m/a²]
    max_ang_vel: 1.0 # Maximal angular velocity in rad/s. Default: 1.5708 [rad/s]
    max_ang_acc: 1.5708 # Maximal angular acceleration in rad/s². Default: 3.1415 [rad/s²]
    decel_ratio: 0.8 # During deceleration the robot uses this partition of its maximal acceleration. Default: 0.8

  grad_mode:
    K: 1.0 # Proportional control parameter. Defualt: 0.8 [-]
    boundary_error: 0.5 # param for calculating translational command velocity in rad. Default: 0.5 [rad]
    max_error: 2.0 # param for calculating translational command velocity in rad. Default: 2/3 PI [rad]
    
  direct_mode:
    min_obst_dist: 14 # If the robot is closer to an obstacle then this -> grad_mode control. Default: 6 [grid cell]
    K: 0.8 # Proportional control parameter. Defualt: 0.8 [-]
    boundary_error: 0.2 # param for calculating translational command velocity in rad. Default: 0.2 [rad]
    max_error: 1.0473 # param for calculating translational command velocity in rad. Default: 1.0473 [rad]
  
  end_mode:
    K: 0.8 # Proportional control parameter. Defualt: 0.8 [-]

  debug:
    publish_occ_grid: true # Publishing the inflated occupancy grid to the topic /grad_field_local_planner/occ_grid
    publish_grad_field: true # Publishing the gradient field to the topic /grad_field_local_planner/grad_field
    publish_des_orient: true # Publishing the desired orientation calculated based on the gradient field.
#				It is always calulated based on the gradient field no matter what, so it is not 
#				representative, when the controller is in direct or end mode. 
